appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -0,0 +1,390 @@
+     * components and setting back all the properties.
+     */
+    private void reinitializeChart() {
+      // Re-set all Chart properties to take effect on
+      // the newly instantiated Chart View
+      for (EditableProperty property : properties) {
+        // The Type property should not be re-set, since
+        // this method call is part of the Type setting process.
+        if (!property.getName().equals(PROPERTY_NAME_TYPE)) {
+          onPropertyChange(property.getName(), property.getValue());
+        }
+      }
+
+      chartView.getChartWidget().draw();
+
+      // Re-attach all children MockChartData components.
+      // This is needed since the properties of the MockChart
+      // are set after the Data components are attached to
+      // the Chart, and thus they need to be re-attached.
+      for (MockComponent child : children) {
+          ((MockChartData) child).addToChart(MockChart.this);
+      }
+    }
+
+    @Override
+    public int getPreferredWidth() {
+        return ComponentConstants.VIDEOPLAYER_PREFERRED_WIDTH;
+    }
+
+    @Override
+    public int getPreferredHeight() {
+        return ComponentConstants.VIDEOPLAYER_PREFERRED_HEIGHT;
+    }
+
+    @Override
+    public void onPropertyChange(String propertyName, String newValue) {
+        super.onPropertyChange(propertyName, newValue);
+
+        if (propertyName.equals(PROPERTY_NAME_TYPE)) {
+            setTypeProperty(newValue);
+            changeChartPropertyVisibilities();
+        } else if (propertyName.equals(PROPERTY_NAME_BACKGROUNDCOLOR)) {
+            chartView.setBackgroundColor(newValue);
+        } else if (propertyName.equals(PROPERTY_NAME_DESCRIPTION)) {
+            chartView.setTitle(newValue);
+            chartView.getChartWidget().draw(); // Title changing requires re-drawing the Chart
+        } else if (propertyName.equals(PROPERTY_NAME_LEGEND_ENABLED)) {
+          setLegendEnabledProperty(newValue);
+        } else if (propertyName.equals(PROPERTY_NAME_GRID_ENABLED)) {
+          setGridEnabledProperty(newValue);
+        } else if (propertyName.equals(PROPERTY_NAME_PIE_RADIUS)) {
+            setPieRadiusProperty(newValue);
+        } else if (propertyName.equals(PROPERTY_NAME_LABELS_FROM_STRING)) {
+          setLabelsFromStringProperty(newValue);
+        }
+    }
+
+    /**
+     * Creates Data components from the contents of the specified MockDataFile component.
+     * The Data components are then attached as children to the Chart and the Source property
+     * of each individual Data component is set accordingly.
+     *
+     * @param dataFileSource  MockDataFile component to instantiate components from
+     */
+    public void addDataFile(MockDataFile dataFileSource) {
+        List<String> columnNames = dataFileSource.getColumnNames();
+
+        for (String column : columnNames) {
+            // Create a new MockChartData2D component and attach it to the Chart
+            // TODO: More data component support
```

Comment:
@ewpatton Mainly if 3D chart data types are ever introduced (or even the Bubble Chart, which takes in a 3rd property of the radius). The Bubble Chart was mentioned briefly in the proposal, but 3D charts were only ever discussed in some meetings, so there's no pointers to it.
Should I document this more thoroughly somewhere?

Reply:
No, I think there's sufficient functionality in this PR that we can just leave this to future work. It might be useful for you to give us a blog post we could put on our website to highlight everything that you did though, so a discussion of what didn't go in could be included there.

Reply:
@ewpatton Fair enough! How should I go about providing a blog post for the website? Should it be in Markdown format, and also how about putting images on the post?

There was a bit of a summary I made here, but for the purposes of MIT App Inventor's site, I figure I could only take bits from this:
https://elatoskinas.github.io/blog/open%20source/2019/08/25/gsoc-2019-project-summary.html